map/src/action.js

```diff
@@ -0,0 +1,366 @@
+const fs = require("fs");
+const core = require("@actions/core");
+const exec = require("@actions/exec");
+const insightsAction = require("./insights");
+
+/**
+ * Asynchronously reads the contents of a file.
+ * @param {string} filePath - The path of the file you want to read.
+ * @returns {Promise<string>} The contents of the file as a string.
+ */
+async function readFile(filePath) {
+  return new Promise((resolve, reject) => {
+    fs.readFile(filePath, { encoding: "utf-8" }, (err, data) => {
+      if (err) {
+        return reject(err);
+      }
+      resolve(data.toString());
+    });
+  });
+}
+
+function setupEnv() {
+  core.setSecret("api_token");
+}
+
+function isPullRequestEvent(githubEventName) {
+  // We currently listen for "pull_request_target" since that will allow secrets
+  // to be passed in for forked repos, but we used to listen for "pull_request",
+  // so we keep it here for backwards compatibility.
+  return (
+    githubEventName === "pull_request" ||
+    githubEventName === "pull_request_target"
+  );
+}
+
+function isForkedPullRequestEvent(githubEventName, githubEventData) {
+  return (
+    isPullRequestEvent(githubEventName) &&
+    githubEventData.pull_request.head.repo.fork
+  );
+}
+
+function isInsecureConfiguration(config) {
+  // This checks to see if we are in an insecure configuration.
+  // If we are running maps, there is a possibility that user-supplied
+  // code may run. This is only possible with Python right now.
+  // If that happens, we want to disable Python so this check allows
+  // us to check if that situation is present.
+  return config.apiToken && config.languages.python;
+}
+
+const INSIGHTS_DISABLED = "Insights Disabled";
+
+async function needsInsights(config) {
+  const args = [
+    "codesee@latest",
+    "metadata",
+    "--repo",
+    config.repoOrigin,
+    "-a",
+    config.apiToken,
+    "-o",
+    "codesee.metadata.json",
+  ];
+
+  if (config.codeseeUrl) {
+    args.push("--url", config.codeseeUrl);
+  }
+
+  const runExitCode = await exec.exec("npx", args);
+  if (runExitCode !== 0) {
+    // If we can't get the metadata, assume that we need insights!
+    return true;
+  }
+
+  try {
+    const output = JSON.parse(
+      await fs.promises.readFile("codesee.metadata.json", "utf-8")
+    );
+    if (output.insightsDisabled) {
+      return INSIGHTS_DISABLED;
+    }
+    return output.insights.length === 0;
+  } catch (e) {
+    core.warning(
+      `\n\n Unable to read metadata for repo, assuming we need insights: ${e.message}`
+    );
+    return true;
+  }
+}
+
+function getBooleanInput(paramName) {
+  return ["true", "True", "TRUE"].includes(process.env[paramName])
+    ? true
+    : false;
+}
+
+function getConfig() {
```

Comment:
This entire function was changed to use `process.env` instead of `core.getInput`